localfs/store.go

```diff
@@ -173,6 +173,10 @@ func (l *LocalStore) List(ctx context.Context, query cloudstorage.Query) (*cloud
 		if md, ok := metadatas[objname]; ok {
 			obj.metadata = md
 		}
+		if (query.StartOffset != "" && obj.name < query.StartOffset) ||
+			(query.EndOffset != "" && obj.name >= query.EndOffset) {
+			continue
+		}
```

Comment:
[NIT] I would move this into `filepath.Walk()` above and avoid adding it to `objects` at all.